pkg/cmd/cli/config/store.go

```diff
@@ -0,0 +1,165 @@
+package config
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/golang/glog"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client/clientcmd"
+	clientcmdapi "github.com/GoogleCloudPlatform/kubernetes/pkg/client/clientcmd/api"
+
+	"github.com/openshift/origin/pkg/cmd/flagtypes"
+)
+
+const (
+	fromKube      = "fromkube"
+	fromOpenShift = "fromopenshift"
+)
+
+// A ConfigStore is the representation of a config from one individual config file. Can be used
+// to persist configs by being explicit about the file to save.
+type ConfigStore struct {
+	Config         *clientcmdapi.Config
+	Path           string
+	providerEngine string
+}
+
+type ConfigStoreNotFoundError struct {
+	msg string
+}
+
+func (e *ConfigStoreNotFoundError) Error() string {
+	return e.msg
+}
+
+func (c *ConfigStore) FromOpenShift() bool {
+	return c.providerEngine == fromOpenShift
+}
+
+func (c *ConfigStore) FromKube() bool {
+	return c.providerEngine == fromKube
+}
+
+// Load a ConfigStore from the explicit path to a config file provided as argument
+// Error if not found.
+func LoadFrom(path string) (*ConfigStore, error) {
+	config, err := tryToLoad(path, fromOpenShift)
+	if err == nil {
+		return config, nil
+	}
+
+	return nil, &ConfigStoreNotFoundError{fmt.Sprintf("Unable to load config file from '%v': %v", path, err.Error())}
+}
+
+// Load a ConfigStore using the priority conventions declared by the ClientConfigLoadingRules.
+// Error if none can be found.
+func LoadWithLoadingRules() (store *ConfigStore, err error) {
+	for _, path := range OpenShiftClientConfigFilePriority() {
+		store, err = tryToLoad(path, fromOpenShift)
+		if err == nil {
```

Comment:
This code still ignores deserialization errors that should be errors.  If a file in the chain is malformed, then the user intended for us to use a file, put values in there, but we don't understand it.  We need to fail in that case.